src/table/cells/selections.jsx

```diff
@@ -17,20 +18,30 @@ const useStyles = makeStyles({
 
 export default function withSelections(CellComponent) {
   const HOC = (props) => {
-    const { cell, selState, selDispatch, className } = props;
+    const { cell, selState, selDispatch, className, rootElement, rowIndex, columnIndex } = props;
 
     const classes = useStyles();
     const handleMouseUp = (evt) => cell.isDim && evt.button === 0 && selectCell(cell, selState, selDispatch, evt);
     const selectionClass = getSelectionClass(cell, selState);
 
-    return <CellComponent {...props} className={`${classes[selectionClass]} ${className}`} onMouseUp={handleMouseUp} />;
+    return (
+      <CellComponent
+        {...props}
+        className={`${classes[selectionClass]} ${className}`}
+        onMouseUp={handleMouseUp}
+        onKeyDown={(evt) => handleKeyPress(evt, rootElement, rowIndex, columnIndex, cell, selState, selDispatch)}
```

Comment:
So this is the part that needs some thinking. I'm not sure we want to place the listener on the cell. A lot of passing arguments and we need to make sure it works with a bunch of different cases. (selections enabled/disabled, column styling etc). If we put the listener on the entire table, we don't have that much information about the cell and have to fetch all cell using `getElementsByClassName` or similar. But I think it would be a simpler solution, then add a onKeyDown on this cell for clicking enter for selections or something. Not sure which approach to take at this point

Reply:
Yeah, right. I was working on selections as well and decided to place the listener on the cell as it is a much easier solution.

Reply:
If selections are enabled, I agree. But we need to consider the case where you want the keyboard navigation and selections disabled.

Reply:
Yeah, I will work on **Cancels Selections** in key selections PR as well. 
And in which case do we want the keyboard navigation enabled and selections disabled? IMO, in edit mode, we want both the keyboard navigation and selections disabled. In analysis mode, we want both enabled. Could you correct me If I am wrong?

Reply:
That is true, but there might be cases outside sense. @Caele what do you think about this?

Reply:
hmm, I'd have to look more closely, but I think:
- only the focused cell should have a listener (which is added dynamically upon enter)
- moving with the keys also moves the listener (no react re-render involved?)
This kind of skips the entire React flow, but would save us lots of listeners and possible re-renders if we can pull it off. But I'm also not very fluent in react...

Reply:
That's an interesting approach. In that case, where would you handle that? Still in selections? or should it be available when selections are disable?

Reply:
In theory it would be handled in a navigation component thing instead, so separate from selections. (As you could have navigation without selections, in snapshot + screenreader mode).
I did check react, and it adds a listener to the document, not each element. (They will change to the root of the react app, but should be the same for us).

As you enter the table, the state should be row 0, col 0, so that cell recives focus. Then we can update and track that as the arrow keys are pressed.
I'm not entirely sure how to best do it, but I think we need to try and avoid re-rendering as you move around. There is really no need as all we do is change the focus().

Reply:
`I did check react, and it adds a listener to the document, not each element. (They will change to the root of the react app, but should be the same for us).`
So when you add a listener for a bunch of elements, they are all bunched into one big listener?

`I'm not entirely sure how to best do it, but I think we need to try and avoid re-rendering as you move around.` I don't think it rerenders with the current solution. If we start adding states, especially on a higher level, then it might update the entire table.